packages/dcos-integration-test/extra/test_dcos_diagnostics.py

```diff
@@ -401,55 +321,36 @@ def test_dcos_diagnostics_report(dcos_api_session):
         assert len(report_response['Nodes']) > 0
 
 
-def _get_bundle_list(dcos_api_session):
-    response = check_json(dcos_api_session.health.get('/report/diagnostics/list/all'))
-    bundles = []
-    for _, bundle_list in response.items():
-        if bundle_list is not None and isinstance(bundle_list, list) and len(bundle_list) > 0:
-            # append bundles and get just the filename.
-            bundles += map(lambda s: os.path.basename(s['file_name']), bundle_list)
-    return bundles
-
-
-@retrying.retry(wait_fixed=2000, stop_max_delay=120000,
-                retry_on_result=lambda x: x is False)
-def wait_for_diagnostics_job(dcos_api_session, last_datapoint):
-    response = check_json(dcos_api_session.health.get('/report/diagnostics/status/all'))
-    # find if the job is still running
-    job_running = False
-    percent_done = 0
-    for _, attributes in response.items():
-        assert 'is_running' in attributes, '`is_running` field is missing in response'
-        assert 'job_progress_percentage' in attributes, '`job_progress_percentage` field is missing in response'
-
-        if attributes['is_running']:
-            percent_done = attributes['job_progress_percentage']
-            logging.info("Job is running. Progress: {}".format(percent_done))
-            job_running = True
-            break
-
-    # if we ran this bit previously compare the current datapoint with the one we saved
-    if last_datapoint['time'] and last_datapoint['value']:
-        if percent_done <= last_datapoint['value']:
-            assert (datetime.datetime.now() - last_datapoint['time']) < datetime.timedelta(seconds=15), (
-                "Job is not progressing"
-            )
-    last_datapoint['value'] = percent_done
-    last_datapoint['time'] = datetime.datetime.now()
+@pytest.mark.supportedwindows
+def test_dcos_diagnostics_bundle_crete_download_delete(dcos_api_session):
+    """
+    test bundle create, read, delete workflow
+    """
+    _create_bundle(dcos_api_session)
+    _check_diagnostics_bundle_status(dcos_api_session)
+    _download_and_extract_bundle(dcos_api_session)
+    _download_and_extract_bundle_from_another_master(dcos_api_session)
+    _delete_bundle(dcos_api_session)
```

Comment:
Future improvement for another PR: These functions seem to work together by coincidence, with the assumption that they're the only things creating diagnostics bundles as they execute. `_create_bundle()` creates a single bundle, but the other functions operate on all available bundles: all available bundles get downloaded, extracted, and deleted, regardless of what created them. This test could be improved by having `_create_bundle()` return the bundle's name and then passing it to the remaining functions, so that they operate on just that bundle.

Reply:
I love this idea. I'll fix it after weekend.